index.bs

```diff
@@ -222,41 +239,63 @@ The following code demonstrates presenting a simple rendering loop to a {{VRDisp
 
 <pre class="lang-js">
 var frameData = new VRFrameData();
+var attachedFrameOfReference;
+var stationaryFrameOfReference;
+
+function drawAttachedGeometry {
+  // Draw UI for 3DOF experience
+}
+
+function drawStationaryGeometry {
+  // Draw UI for standing experience
+}
 
 // Render a single frame of VR data.
 function onVRFrame() {
   // Schedule the next frame's callback
   vrDisplay.requestAnimationFrame(onVRFrame);
+    
+  gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
 
-  // Poll the VRDisplay for the current frame's matrices and pose
-  vrDisplay.getFrameData(frameData);
+  var drawFunction;
 
-  gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+  // Attempt to draw world-locked content
+  if (stationaryFrameOfReference && vrDisplay.getFrameData(stationaryFrameOfReference.coordinateSystem, frameData)) {
+    drawFunction = drawStationaryGeometry;
+  }
+  else {
+    // This must always succeed
+    vrDisplay.getFrameData(attachedFrameOfReference.coordinateSystem, frameData);
+    drawFunction = drawStageGeometry;
+  }
 
   // Render to the left eye's view to the left half of the canvas
   gl.viewport(0, 0, canvas.width * 0.5, canvas.height);
   gl.uniformMatrix4fv(projectionMatrixLocation, false, frameData.leftProjectionMatrix);
   gl.uniformMatrix4fv(viewMatrixLocation, false, frameData.leftViewMatrix);
-  drawGeometry();
+  drawFunction();
 
   // Render to the right eye's view to the right half of the canvas
   gl.viewport(canvas.width * 0.5, 0, canvas.width * 0.5, canvas.height);
   gl.uniformMatrix4fv(projectionMatrixLocation, false, frameData.rightProjectionMatrix);
   gl.uniformMatrix4fv(viewMatrixLocation, false, frameData.rightViewMatrix);
-  drawGeometry();
+  drawFunction();
 
   // Indicate that we are ready to present the rendered frame to the VRDisplay
   vrDisplay.submitFrame();
```

Comment:
So here's a fun question, though it seems Windows Holographic must already handle this: I render my scene using N different CoordinateSystems. Maybe a couple of stationary ones for objects around my play space and an attached one for the UI because reasons. When it comes time to call `submitFrame` what do I feed back to the VR API for reprojection purposes?

I think it actually works out fine on most devices that I'm familiar with, because they won't actually be reporting N different poses. They'll report one internally and the various CoordinateSystems will just be transforms from that, so the one internal one will be what's used as the basis for reprojection. Do we feel like that'll hold true for every device type, though?